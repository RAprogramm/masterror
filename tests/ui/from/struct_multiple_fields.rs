// SPDX-FileCopyrightText: 2025 RAprogramm <andrey.rozanov.vl@gmail.com>
//
// SPDX-License-Identifier: MIT

use masterror::Error;

#[derive(Debug, Error)]
#[error("{left:?} - {right:?}")]
struct BadStruct {
    #[from]
    left: DummyError,
    #[backtrace]
    trace: Option<std::backtrace::Backtrace>,
    right: DummyError,
}

#[derive(Debug, Error)]
#[error("dummy")] 
struct DummyError;

fn main() {}
