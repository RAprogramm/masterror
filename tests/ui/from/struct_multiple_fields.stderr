error: deriving From requires no fields other than source and backtrace
 --> tests/ui/from/struct_multiple_fields.rs:6:5
  |
6 |     #[from]
  |     ^^^^^^^
